### DISCUSSION TITLE
Multicam

### DISCUSSION THREAD

**Discussion by tdashmike** [gtxaspec/wz_mini_hacks#535] [Q&A] [answered]
How do I get the multicam to work on the webwerver? It just shows a bunch of blank blocks. Thank you.

**Comment by Bob-AOL** [comment]
@virmaior

**Comment by virmaior** [comment, marked as answer]
I've actually made some revisions to the code since the most recent version (https://github.com/virmaior/wz_mini_hacks)

to get it to work, the key is 
```
window.onload = function()
{
  cam_tool.compose(5,'/z%d/cgi-bin/jpeg.cgi?channel=0');
  cam_tool.add_camera(6,'/z7/cgi-bin/get_jpeg.cgi?base=0');
  cam_tool.init();
};
```

to add a camera by itself:
`cam_tool.add_camera(1,'http://yourip/cgi-bin/jpeg.cgi?channel=0');`

conversely `compose` builds them according to a pattern.

In my case, what I'm doing involves an RPI and a reverse proxy so all of them look like:

http://rpi.local/z%d/jpeg.cgi?channel=0
=  http://rpi.local/z1/jpeg.cgi?channel=0  ,  http://rpi.local/z2/jpeg.cgi?channel=0    ...

the format for my sixth camera is different because it's a AtomCam rather than Wyze.


---
**Background assumptions:**
make sure that you're using a version of the firmware where you can still get jpeg.cgi to work.

**Reply by tdashmike** [reply]
Thanks! I got it to work.

```
window.onload = function()
{
  cam_tool.add_camera(1,'http://192.168.xxx.xxx/cgi-bin/jpeg.cgi?channel=0');
  cam_tool.add_camera(2,'http://192.168.xxx.xxx/cgi-bin/jpeg.cgi?channel=0');
  cam_tool.add_camera(3,'http://192.168.xxx.xxx/cgi-bin/jpeg.cgi?channel=0');
  cam_tool.add_camera(4,'http://192.168.xxx.xxx/cgi-bin/jpeg.cgi?channel=0');
  cam_tool.init();
};
```

By the way, I will be in Tokyo in August.

**Reply by virmaior** [reply]
nice. if you use the latest version of this page (not yet incorporated in gtxaspec's) it will give you:

1. average time to load for each image
2. red outline on fail
3. experimental option to auto-guess how frequently one can update.

e.g., in my use case, when at home I do 1 image / 500 ms as the default vs. when away at 1 image / 1.5 seconds (the only server I control is in the US, so VPN to us, us website, back to Japan)

**Reply by virmaior** [reply]
new version waiting to get pulled in (https://github.com/gtxaspec/wz_mini_hacks/pull/627).

if you update all of your cameras to the latest wz_mini it will now display day/night info for each cam. I also split out the html / css /js. For composing with your cameras, you just need to keep those lines for the main one. Also the interval is now part of `cam_tool.first_init()`
right now on my local I do an 800ms cycle for 6 cameras and when remote a 3000ms cycle for 5 cameras.

e.g., my current code:
```
<script type="text/javascript">

window.onload = function()
{
  cam_tool.compose(5,'/z%d/cgi-bin/jpeg.cgi?channel=0','v3');
  cam_tool.add_camera(6,'/z7/cgi-bin/get_jpeg.cgi?base=0');
  cam_tool.init(800);
};
</script>

```

"v3" at the end of the compose or as a third parameter for add_camera will tell it that the camera should support the test_night();
